### PR TITLE
Add production support for workspaces

### PR DESCRIPTION
**Pull request by nicknytko**
Todo:

- [x] Fix permissions so that jupyter and vscode can actually create files in their home directories
- [x] ~S3 buckets don't seem to be syncing yet~ (this was a silly config error :facepalm: )

### CODE REVIEW

File: lib/aws.js [hunk 1]
```diff
@@ -26,3 +27,20 @@ module.exports.init = function(callback) {
...
     callback(null);
 };
+
+module.exports.isOnEC2 = function(callback) {
+    /* Try to connect to the AWS Metadata service, if we get a host not found then
+       we are very likely to be running on EC2 */
+    request(`http://${AWS.MetadataService.host}`, (err, _response, _body) => {
+        callback(null, !err);
+    });
+};
```
Comment (nicknytko) [inline lib/aws.js:37]: this can be replaced with the config flag for running in ec2

Comment (nicknytko) [reply lib/aws.js:37]: `config.runningInEc2`

File: lib/config.js [hunk 2]
```diff
@@ -113,8 +113,9 @@ config.filesRoot = '/files';
 config.workspaceS3Bucket = 'workspaces';
 config.workspaceDevHostInstanceId = 'devHost1';
 config.workspaceDevHostHostname = 'localhost';
-config.workspaceDevHostPort = 8081;
+config.workspaceHostPort = 8081;
 config.workspaceDevContainerHostname = 'host.docker.internal';
+config.workspaceJobsDirectory = '/jobs';
```
Comment (tdy) [inline lib/config.js:118]:
```suggestion
config.workspaceJobsDirectory = process.cwd();
```
Should this be `cwd`? That was the original default.

File: workspace_host/interface.js [hunk 3]
```diff
@@ -589,52 +655,78 @@ function _createContainer(workspace_id, port, settings, callback) {
...
-        port_id_mapper[port] = workspace_id;
-        sqldb.query(sql.update_load_count, {workspace_id, count: +1}, function(err, _result) {
+        (callback) => {
+            docker.createContainer({
+                Image: settings.workspace_image,
+                ExposedPorts: {
+                    [`${settings.workspace_port}/tcp`]: {},
+                },
+                Env: [
+                    `WORKSPACE_BASE_URL=/workspace/${workspace_id}/container/`,
```
Comment (nicknytko) [inline workspace_host/interface.js:691]:
```suggestion
                    `WORKSPACE_BASE_URL=/pl/workspace/${workspace_id}/container/`,
```

**Pull request merged**